src/parser/processor.ts

```diff
@@ -0,0 +1,81 @@
+import * as fs from "fs";
+import configuration from "../configuration/config-reader";
+import { GetActivity } from "../get-activity";
+import program from "./command-line";
+
+export class Processor {
+  private _transformers: Array<Transformer> = [];
+  private _result: Result = {};
+  private readonly _configuration = configuration;
+
+  add(transformer: Transformer) {
+    this._transformers.push(transformer);
+    return this;
+  }
+  async run(data: Readonly<GetActivity>) {
+    if (this._configuration.disabled) {
+      console.log("Module is disabled. Skipping...");
+      return;
+    }
+    for (const transformer of this._transformers) {
+      if (transformer.enabled) {
+        this._result = await transformer.transform(data, this._result);
+      }
+    }
+    // Aggregate total result
+    for (const item of Object.keys(this._result)) {
+      this._result[item].total = this._sumRewards(this._result[item]);
+    }
+    return this._result;
+  }
+  dump() {
+    const { file } = program.opts();
+    const result = JSON.stringify(this._result, undefined, 2);
+    if (!file) {
+      console.log(result);
+    } else {
+      fs.writeFileSync(file, result);
+    }
+  }
+  _sumRewards(obj: Record<string, object | number>) {
+    let totalReward = 0;
+
+    for (const [key, value] of Object.entries(obj)) {
+      if (Object.prototype.hasOwnProperty.call(obj, key)) {
+        if (key === "reward" && typeof value === "number") {
+          totalReward += value;
+        } else if (typeof value === "object") {
+          totalReward += this._sumRewards(value as Record<string, object>);
+        }
+      }
+    }
+
+    return totalReward;
+  }
+}
+
+export interface Transformer {
+  transform(data: Readonly<GetActivity>, result: Result): Result | Promise<Result>;
+  get enabled(): boolean;
+}
+
+export interface Result {
+  [k: string]: {
+    comments?: Array<Comment>;
```

Comment:
Perhaps its best to initialize as an empty array instead?
```suggestion
    comments: Array<Comment>;
```

Reply:
The point was that if you disable the comment module, you will just end up with empty arrays everywhere, which might not be very useful

Reply:
This whole thing is the comment module. That may make sense only from the perspective of the kernel which is a seperate codebase?

Reply:
I thought it would be nice to be very flexible, is there is a use case for example an organization would like to reward bounties without considering the comment (just reward the base price) that is possible through the configuration